.github/actions/gem-dependency-updater/scripts/gem_dependency_updater.rb

```diff
@@ -0,0 +1,125 @@
+    create_pull_request
+  end
+
+  private
+
+  def is_merge_event?
+    return false unless @github_event['action'] == 'closed'
+    return false unless @github_event['pull_request']
+
+    @github_event['pull_request']['merged'] == true
+  end
+
+  def validate_arguments
+    abort(USAGE_MESSAGE) if @branch_name.nil? || @branch_name.strip.empty? || @gem_name.nil? || @gem_name.strip.empty?
+  end
+
+  def checkout_branch
+    execute_command("git fetch --depth=1 origin", "Failed to fetch from origin.")
+    checkout_cmd = <<~CMD
+      git checkout #{@dependent_repo_branch_name} 2>/dev/null \
+        || git checkout -b #{@dependent_repo_branch_name} origin/#{@dependent_repo_branch_name} 2>/dev/null \
+        || git checkout -b #{@dependent_repo_branch_name}
+    CMD
+    execute_command(checkout_cmd, "Failed to checkout or create branch '#{@dependent_repo_branch_name}'.")
+    execute_command("git pull", "Failed to pull latest changes for branch '#{@dependent_repo_branch_name}'.", graceful: true)
+  end
+
+  def update_gem_dependency
+    execute_command("bundle config set --local frozen false", "Failed to set bundle config.")
+    gemfile = File.read(GEMFILE_PATH)
+    new_gemfile = gemfile.gsub(/(gem '#{Regexp.escape(@gem_name)}',.*branch: )'[^']*'/) do
+      "#{$1}'#{@branch_name}'"
+    end
+    File.write(GEMFILE_PATH, new_gemfile)
+    execute_command("bundle update #{@gem_name} --conservative", "Failed to update gem dependency.")
+  end
+
+  def commit_and_push_changes
+    configure_git_user
+    execute_command('git add Gemfile Gemfile.lock')
+    commit_message = "Update #{@gem_name} to feature branch #{@branch_name}"
+    output = execute_command("git commit -m \"#{commit_message}\"", "Failed to commit changes.", graceful: true)
+
+    if output.include?('nothing to commit, working tree clean')
+      puts "Nothing to commit, working tree clean."
+      return
+    end
+
+    execute_command("git push origin #{@dependent_repo_branch_name}")
+  end
+
+  def create_pull_request
+    create_pr_command = [
+      "gh pr create",
+      "--title \"Update #{@gem_name} to branch #{@branch_name}\"",
+      "--body \"This PR updates the #{@gem_name} to the latest feature branch.\"",
+      "--head #{@dependent_repo_branch_name}",
+      "--base #{BASE_BRANCH}"
+    ].join(' ')
+
+    puts "Creating pull request for branch '#{@branch_name}'..."
+
+    output = execute_command(create_pr_command, "Failed to create pull request.", graceful: true)
+
+    if output.include?('already exists')
+      puts "Pull request already exists for branch '#{@branch_name}'."
+      return
+    end
+  end
+
+  def configure_git_user
+    execute_command("git config user.name \"#{GIT_USER_NAME}\"")
+    execute_command("git config user.email \"#{GIT_USER_EMAIL}\"")
+  end
+
+  def execute_command(command, error_message = nil, graceful: false)
+    puts "Executing: #{command}"
+    output = `#{command} 2>&1`
+
+    unless $?.success?
```

Comment:
Wow! I am seeing this `$?` for the first time! It's like proper raw system Ruby 😍